components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -17761,6 +17762,736 @@ public void deleteTenantTheme(int tenantId) throws APIManagementException {
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            connection.setAutoCommit(false);
+            try {
+                if (DevPortalConstants.PUBLISH.equals(action)) {
+                    String draftedArtifact = getDraftedArtifactForOrg(connection, organization);
+                    if (draftedArtifact != null) {
+                        InputStream artifactContent = getArtifactContent(connection, draftedArtifact);
+                        String newUUID = addArtifact(connection, artifactContent, DevPortalConstants.PUBLISHED_ORG_THEME);
+                        updatePublishedArtifactForOrg(connection, organization, newUUID);
+                        removeArtifact(connection, draftedArtifact);
+                    } else {
+                        log.warn("ID cannot be found in drafted state. Aborting the request");
+                    }
+                } else {
+                    String publishedArtifact = getPublishedArtifactForOrg(connection, organization);
+                    if (publishedArtifact != null) {
+                        InputStream artifactContent = getArtifactContent(connection, publishedArtifact);
+                        String newUUID = addArtifact(connection, artifactContent, DevPortalConstants.DRAFTED_ORG_THEME);
+                        updateDraftedArtifactForOrg(connection, organization, newUUID);
+                        removeArtifact(connection, publishedArtifact);
+                    } else {
+                        log.warn("ID cannot be found in published state. Aborting the request");
+                    }
+                }
+                connection.commit();
+            } catch (SQLException e) {
+                connection.rollback();
+                handleException("Failed to update organization theme status for organization " + organization, e);
+            }
+        } catch (SQLException e) {
+            handleException("Database connection error while updating organization theme status for organization " + organization, e);
+        }
+    }
+
+
+    /**
+     * Deletes an organization theme.
+     *
+     * @param organization Organization name.
+     * @param themeId      Theme ID to delete.
+     * @throws APIManagementException If a database error occurs.
+     */
+    public void deleteOrgTheme(String organization, String themeId) throws APIManagementException {
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            connection.setAutoCommit(false);
+            try {
+                if (isThemeUsedByOrg(connection, organization, themeId)) {
+                    removeArtifact(connection, themeId); // Due to DB rules, foreign ID will also be set to NULL
+                    removeOrgIfNoData(connection, organization);
+                    connection.commit();
+                } else {
+                    log.warn("User's organization does not use the theme. Aborting the request");
```

Comment:
If the theme has already in use, we should indicate that in the UI and abort the action. But how that messaging is populated to the UI?

Reply:
+1. To populate, we need a proper response for APIs. Hence add the changes to throw exceptions by [6th set of required changes](https://github.com/wso2/carbon-apimgt/pull/12874/commits/a6d6cd8b552949be6ea2ae144a5bfc530dc4eaf8)